src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -102,6 +102,10 @@ public function preSubmit(FormEvent $event)
         $form = $event->getForm();
         $data = $event->getData();
 
+        if ($data instanceof \Traversable) {
+            @trigger_error('Support for Traversable is deprecated since version 3.1 and will be removed in 4.0. You should not use it anymore.', E_USER_DEPRECATED);
```

Comment:
It may be OK for data init as array, but an object implementing only `\Traversable` would trigger the notice but was not supported before unless supporting `\ArrayAccess` too.

What about `'Support for objects implementing both \Traversable and \ArrayAccess is deprecated since version 3.1 and will be removed in 4.0. You should not use it anymore.'` ?